### DISCUSSION TITLE
Using Spektral for Link Prediction

### DISCUSSION THREAD

**Discussion by dimmykarson** [danielegrattarola/spektral#128] [Q&A] [answered]
Hi Daniele, 

Could I use spektral for Link Prediction? For example:

G = (V, E)
V = {"A", "B", "C", "D"}
E =  {(A, B, "label 1"), (A, C, "label 1"), (C, D, "label 2"), (B, D, "label 2"),  }//E is a set of weighted edges. 

I wanna predict the label for edge (B, C).

**Comment by danielegrattarola** [comment, marked as answer]
You most definitely can! 
Usually, it involves some layer for learning node representations (e.g., a couple of GCNConvs) and then computing a similarity between the learned representations. You can use `spektral.layers.InnerProductDecoder` for computing a simple inner product between all your nodes and get a sort of predicted adjacency matrix. 

There isn't a pre-made example yet in Spektral, but all the building blocks are there. 
You can start here: https://arxiv.org/abs/1611.07308

Cheers
